lizmap/modules/lizmap/lib/Request/Proxy.php

```diff
@@ -21,6 +21,94 @@ class Proxy
      */
     protected static $_profiles = array();
 
+    protected static $services = null;
+
+    protected static $appContext = null;
+
+    public static function setServices($services = null)
+    {
+        if (!self::$services) {
+            if ($services) {
+                self::$services = $services;
+            } else {
+                self::$services = \lizmap::getServices();
+            }
+        }
+
+        return self::$services;
+    }
+
+    public static function setAppContext($appContext = null)
```

Comment:
same as setServices: we should have a `getAppContext()` and a `setAppContext($appContext)`